pytorch_lightning/plugins/training_type/parallel.py

```diff
@@ -138,3 +138,6 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+        # Un-reference the wrapper if any was used.
+        self.model = self.lightning_module
```

Comment:
Putting this before the `on_gpu` block above means that the `empty_cache` call could also apply to the memory released by de-allocating the wrapper

Reply:
Yes, it seems `self.model = self.lightning_module` breaks multiple plugins. We will need to investigated deeper.